src/security/ServerOptions.cc

```diff
@@ -50,7 +54,18 @@ Security::ServerOptions::parse(const char *token)
     }
 
     // parse the server-only options
-    if (strncmp(token, "dh=", 3) == 0) {
+    if (strncmp(token, "context=", 8) == 0) {
+#if USE_OPENSSL
+        staticContextSessionId = SBuf(token+8);
+        if (staticContextSessionId.length() > SSL_MAX_SSL_SESSION_ID_LENGTH) {
+            debugs(83, DBG_CRITICAL, "FATAL: Option 'context=' value is too long. Maximum " << SSL_MAX_SSL_SESSION_ID_LENGTH << " characters.");
+            self_destruct();
+        }
+#else
+        debugs(83, DBG_PARSE_NOTE(DBG_IMPORTANT), "WARNING: Option 'context=' requires --with-openssl. Ignoring.");
```

Comment:
I do not consider this particular problem important. If you insist on "hiding" the context ID value, then I am not going to block this PR because of the resulting minor code duplication. Please add a comment to document the rationale:

```
// to hide its arguably sensitive value, do not print `token` here
```